pkg/internal/codegen/parse/util.go

```diff
@@ -231,6 +231,17 @@ func hasSingular(t *types.Type) bool {
 	return false
 }
 
+// hasRequired returns true if t is annotated with
+// +required
+func hasRequired(t *types.Type) bool {
+	for _, c := range t.CommentLines {
+		if strings.Contains(c, "+required"){
```

Comment:
this should be more precize. Is this the same way that `+optional` uses now? I would have expected `strings.Trim(c) == "// +required`.

Reply:
So `+optional` actually isn't implemented at all right now, and those comments we just added don't do anything. It was proposed in https://github.com/kubernetes-sigs/controller-tools/issues/137#issue-405852132, and if you look at the proposed change it does use `TrimLeft` but I just went with `Contains` again to keep with the other util methods in here. 

If you think trim is better we can make that change, but this is what all the other tags use now

Reply:
I just want to avoid that people write `//    +required`. We should be strict and potentially warn otherwise.